src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitString.Fixer.cs

```diff
@@ -0,0 +1,111 @@
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Operations;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public class PreferConstCharOverConstUnitStringFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(PreferConstCharOverConstUnitStringAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document doc = context.Document;
+            CancellationToken cancellationToken = context.CancellationToken;
+            SyntaxNode root = await doc.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            if (root.FindNode(context.Span) is SyntaxNode expression)
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        title: MicrosoftNetCoreAnalyzersResources.PreferConstCharOverConstUnitStringInStringBuilderMessage,
+                        createChangedDocument: async c =>
+                        {
+                            SemanticModel semanticModel = await doc.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                            var operation = semanticModel.GetOperationWalkingUpParentChain(expression, cancellationToken);
+                            if (operation is IVariableDeclaratorOperation variableDeclaratorOperation)
+                            {
+                                return await HandleVariableDeclarator(variableDeclaratorOperation, doc, root, cancellationToken).ConfigureAwait(false);
+                            }
+                            else if (operation is IArgumentOperation argumentOperation)
+                            {
+                                if (argumentOperation.Value is ILiteralOperation argumentLiteral)
+                                {
+                                    return await HandleStringLiteral(argumentLiteral, doc, root, cancellationToken).ConfigureAwait(false);
+                                }
+                            }
+                            return doc;
+                        },
+                        equivalenceKey: "PreferConstCharOverConstUnitStringInStringBuilderAppend"),
+                    context.Diagnostics);
+            }
+        }
+
+        private static async Task<Document?> HandleStringLiteral(ILiteralOperation argumentLiteral, Document doc, SyntaxNode root, CancellationToken cancellationToken)
+        {
+            if (argumentLiteral.ConstantValue.HasValue && argumentLiteral.ConstantValue.Value is string unitString && unitString.Length == 1)
```

Comment:
Can we create a static helper method in the analyzer which takes a `ConstantValue` parameter that has this check? It can be shared between different cases within the analyzer and also between analyzer and code fixer.

Reply:
With the new changes, this `if` block is used only in the analyzer now. Not sure it makes sense to put it into a helper anymore?